api/v2/utils.go

```diff
@@ -308,36 +308,33 @@ func (api *API) getCaptchaKey() string {
 // dualSwarmUpload enables uploading data to 2 different swarm nodes at once
 // to provide high-availability of stored data
 func (api *API) dualSwarmUpload(data []byte, isTar bool) (string, error) {
-	resp1, err := api.swarm.endpoint1.Send(swampi.SingleFileUpload, bytes.NewReader(append(data[0:0:0], data...)), map[string][]string{
-		"content-type": {swampi.SingleFileUpload.ContentType(isTar)},
-	})
-	if err != nil {
-		return "", err
-	}
-	responseData, err := ioutil.ReadAll(resp1.Body)
-	if err != nil {
-		resp1.Body.Close()
-		return "", err
-	}
-	hash1 := string(responseData)
-	resp1.Body.Close()
-	if api.swarm.endpoint2 != nil {
-		resp2, err := api.swarm.endpoint2.Send(swampi.SingleFileUpload, bytes.NewReader(append(data[0:0:0], data...)), map[string][]string{
+	var hashes []string
+	for _, endpoint := range api.swarmEndpoints {
+		resp, err := endpoint.Send(swampi.SingleFileUpload, bytes.NewReader(append(data[0:0:0], data...)), map[string][]string{
```

Comment:
I'm not sure why `append(data[0:0:0], data...` isn't just data? I think the extra data copy is unnecessary.

Reply:
Good catch, I thought the bytes reader would drain the data but it doesnt